Update release guide (#192)
precice/python-bindings: docs/ReleaseGuide.md

    * `CHANGELOG.md` on `python-bindings-v2.1.1.1`.
    * There is no need to bump the version anywhere else, since we use the [python-versioneer](https://github.com/python-versioneer/python-versioneer/) for maintaining the version everywhere else.

4. [Draft a New Release](https://github.com/precice/python-bindings/releases/new) in the `Releases` section of the repository page in a web browser.

    * The release tag needs to be the exact version number (i.e.`v2.1.1.1` or `v2.1.1.1rc1`, compare to [existing tags](https://github.com/precice/python-bindings/tags)).

>     * If this is a full release, use `@target:master`. If this is a pre-release, use `@target:python-bindings-v2.1.1.1`. If you are making a pre-release, **directly skip to the [pre-release](#pre-release) section below**.

Reviewer: ```suggestion
    * If this is a stable release, use `@target:master`. If this is a pre-release, use `@target:python-bindings-v2.1.1.1`. If you are making a pre-release, **directly skip to the [pre-release](#pre-release) section below**.
```
Better wording? https://en.wikipedia.org/wiki/Software_release_life_cycle

    * Release title is also the version number (i.e. `v2.1.1.1` or `v2.1.1.1rc1`, compare to [existing releases](https://github.com/precice/python-bindings/tags)).

5. As soon as one approving review is made, merge the release PR (from `python-bindings-v2.1.1.1`) into `master`.

6. Merge `master` into `develop` for synchronization of `develop`.

> 7. If everything is in order up to this point then the new version can be released by hitting the "Publish release" button in your Release Draft. This will create the corresponding tag and trigger [publishing the release to PyPI](https://github.com/precice/python-bindings/actions?query=workflow%3A%22Upload+Python+Package%22).

Reviewer: ```suggestion
7. If everything is in order up to this point then the new version can be released by hitting the "Publish release" button in your release Draft. This will create the corresponding tag and trigger [publishing the release to PyPI](https://github.com/precice/python-bindings/actions?query=workflow%3A%22Upload+Python+Package%22).
```

> 8. Add an empty commit on master by running the steps:

Reviewer: ```suggestion
8. Add an empty commit (details https://github.com/precice/python-bindings/issues/109) on master by running the steps:
```

   For more details refer to https://github.com/precice/python-bindings/issues/109 and https://github.com/python-versioneer/python-versioneer/issues/217.

9. *Temporarily not maintained* Update Spack package (refer to `python-bindings/spack/README.md`).

## Pre-release

> A pre-release branch is never merged into a master branch. After creating the branch and drafting a release, directly hit the "Publish release" button in your Release Draft. Now you can check the pre-release artifacts (e.g. release on [PyPI](https://pypi.org/project/pyprecice/#history)) of the release. No further action is required for a pre-release.

Reviewer: ```suggestion
The purpose of a pre-release is to test the release workflow (packaging, upload to PyPI). After drafting a release, directly hit the "Publish release" button in your Release Draft. Now you can check the pre-release artifacts (e.g. release on [PyPI](https://pypi.org/project/pyprecice/#history)) of the release. No further action is required for a pre-release.
```

Is this actually true? Aren't we just testing the release workflow with a pre-release? We can actually create a pre-release from the release branch that will later be merged into master. This makes perfectly sense.

Author: Creating a pre-release from the release branch itself makes sense, we should do this for the future. But until now we never did this.
